custom/icds_reports/static/icds_reports/js/spec/progress-report.directive.spec.js

```diff
@@ -0,0 +1,209 @@
+        chai.expect(controller).not.to.be.a('undefined');
+    });
+
+    it('tests initial state', function () {
+        assert.equal(controller.showWarning, true);
+        assert.deepEqual(controller.filtersData, {});
+    });
+
+    it('tests show info message', function () {
+        var fakeDate = new Date(2016, 1, 1);
+        var clock = sinon.useFakeTimers(fakeDate.getTime());
+
+        var expected = true;
+        var result = controller.showInfoMessage();
+
+        assert.equal(expected, result);
+        clock.restore();
+    });
+
+    it('tests not show info message', function () {
+        var fakeDate = new Date(2016, 1, 2);
+        var clock = sinon.useFakeTimers(fakeDate.getTime());
+
+        var expected = false;
+        var result = controller.showInfoMessage();
+
+        assert.equal(expected, result);
+        clock.restore();
+    });
+
+    it('tests location change', function () {
+        controller.loadData();
+        controller.selectedLocations.push(
+            {name: 'name1', location_id: 'test_id1'},
+            {name: 'name2', location_id: 'test_id2'},
+            {name: 'name3', location_id: 'test_id3'},
+            {name: 'name4', location_id: 'test_id4'},
+            {name: 'name5', location_id: 'test_id5'},
+            {name: 'name6', location_id: 'test_id6'}
+        );
+        $scope.$digest();
+
+        assert.equal($location.search().location_id, 'test_id4');
+        assert.equal($location.search().selectedLocationLevel, 3);
+        assert.equal($location.search().location_name, 'name4');
+    });
+
+    it('tests moveToLocation national', function () {
+        controller.moveToLocation('national', -1);
+
+        var searchData = $location.search();
+
+        assert.equal(searchData.location_id, '');
+        assert.equal(searchData.selectedLocationLevel, -1);
+        assert.equal(searchData.location_name, '');
+    });
+
+    it('tests moveToLocation not national', function () {
+        controller.moveToLocation({location_id: 'test-id', name: 'name'}, 3);
+
+        var searchData = $location.search();
+
+        assert.equal(searchData.location_id, 'test-id');
+        assert.equal(searchData.selectedLocationLevel, 3);
+        assert.equal(searchData.location_name, 'name');
+    });
+
+    it('tests disable locations for user', function () {
+        controller.userLocationId = 'test_id4';
+        controller.location = {name: 'name4', location_id: 'test_id4'};
+        controller.selectedLocations.push(
+            {name: 'name1', location_id: 'test_id1'},
+            {name: 'name2', location_id: 'test_id2'},
+            {name: 'name3', location_id: 'test_id3'},
+            {name: 'name4', location_id: 'test_id4'},
+            {name: 'name5', location_id: 'test_id5'},
+            {name: 'name6', location_id: 'test_id6'}
+        );
+        var index = controller.getDisableIndex();
+        assert.equal(index, 3);
+    });
```

Comment:
It looks like a lot of these tests are copy/pasted from `awc-reports.directive.spec.js`. would it make sense to combine them or are there chances that they would change in different ways so as to make that not useful?

am not sure how general this question is - but have the same question for any code/tests that have been copy/pasted without modification between these files

Reply:
Some tests are the same because we have similar directives/controllers functionality in a few files.